src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2089,6 +2108,196 @@ SubquerySqlTaskList(Job *job, PlannerRestrictionContext *plannerRestrictionConte
 
 
 /*
+ * ErrorIfUnsupportedJoinReferenceTable errors out if there exists a outer join
+ * exist between reference table and distributed tables.
+ */
+static void
+ErrorIfUnsupportedJoinReferenceTable(PlannerRestrictionContext *plannerRestrictionContext)
+{
+	List *joinRestrictionList =
+		plannerRestrictionContext->joinRestrictionContext->joinRestrictionList;
+	ListCell *joinRestrictionCell = NULL;
+
+	foreach(joinRestrictionCell, joinRestrictionList)
+	{
+		JoinRestriction *joinRestriction = (JoinRestriction *) lfirst(
+			joinRestrictionCell);
+		JoinType joinType = joinRestriction->joinType;
+		PlannerInfo *plannerInfo = joinRestriction->plannerInfo;
+		RelOptInfo *innerrel = joinRestriction->innerrel;
+		RelOptInfo *outerrel = joinRestriction->outerrel;
+
+		switch (joinType)
+		{
+			case JOIN_SEMI:
+			{
+				if (ReferenceTableExist(plannerInfo, outerrel))
+				{
+					ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
```

Comment:
Hmm, the error message might be hard users to understand (i.e, `left-part`). A suggestion:

```cannot plan subqueries involving reference tables on the left tree of a semi-join``` ?

Do you have other suggestions?